File: HoneyStoreUI/src/app/models/order.ts
export class Order {
    id: number;
    userId: number | undefined;    
    firstName: string | undefined;
    lastName: string | undefined;
    phoneNumber: string | undefined;
    email: string | undefined;
    address: string | undefined;
    details: string | undefined;
    deliveryMethod: string | undefined;
    paymentMethod: string | undefined;
    cartItemIds: number[];
    productIds: number[];
    status: string;
    createdOn: Date | undefined;

    constructor(userId: number = 1) {
        this.id = 0;
        this.firstName = '';        
        this.lastName = '';
        this.phoneNumber = '';
        this.email = '';
        this.userId = userId;
        this.cartItemIds = [];
        this.productIds = [];
        this.status = '';
        this.createdOn = new Date(Date.now());
    }
}

File: HoneyStore.DataAccess/Entities/Order.cs
using HoneyStore.DataAccess.Identity;
using HoneyStore.DataAccess.Interfaces;

namespace HoneyStore.DataAccess.Entities
{
    public class Order : IIdentifier
    {
        public int Id { get; set; }

        public int? UserId { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Email { get; set; }

        public string PhoneNumber { get; set; }

        public string Address { get; set; }

        public string Details { get; set; }

        public string DeliveryMethod { get; set; }

        public string PaymentMethod { get; set; }

        public User User { get; set; }

        public DateTime CreatedOn { get; set; }

        public OrderStatus Status { get; set; }

        public ICollection<CartItem> CartItems { get; set; }
    }
}